test/e2e/activator_test.go

```diff
@@ -124,3 +128,105 @@ func TestActivatorOverload(t *testing.T) {
 	}
 	group.Wait()
 }
+
+func TestActivatorRevisionTimeout(t *testing.T) {
+	t.Parallel()
+	cancel := logstream.Start(t)
+	defer cancel()
+
+	const (
+		// The number of concurrent requests to hit the activator with.
+		concurrency = 1
+		// How long the service will process the request in ms.
+		serviceSleep = 10000
+		// body of timeout response from activator
+		wantTimeoutRespBody = "activator request timeout"
+	)
+	var (
+		group       sync.WaitGroup
+		resp        []*spoof.Response
+		mux         sync.Mutex
+		timeoutResp = 0
+		successResp = 0
+	)
+	clients := Setup(t)
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "timeout",
+	}
+
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	defer test.TearDown(clients, names)
+
+	t.Log("Creating a service with run latest configuration.")
+	// Create a service with concurrency 1 that sleeps for N ms.
+	// Limit its maxScale to 10 containers, wait for the service to scale down and hit it with concurrent requests.
+	resources, err := v1test.CreateServiceReady(t, clients, &names,
+		rtesting.WithContainerConcurrency(1),
+		rtesting.WithRevisionTimeoutSeconds(15),
+		rtesting.WithConfigAnnotations(map[string]string{
+			autoscaling.TargetBurstCapacityKey: "-1",
+			autoscaling.MaxScaleAnnotationKey:  "1",
+		}))
+	if err != nil {
+		t.Fatal("Unable to create resources:", err)
+	}
+
+	deploymentName := rnames.Deployment(resources.Revision)
+	if err := WaitForScaleToZero(t, deploymentName, clients); err != nil {
```

Comment:
Is it necessary for it to scale to zero?

Reply:
thanks for pointing it out. Addressed it